### DISCUSSION TITLE
Is there a health check API for port 9345 of rke2 server?

### DISCUSSION THREAD

**Discussion by huangyutongs** [rancher/rke2#6758] [Q&A] [answered]
I want to provide a highly available rke2 server registration endpoint, so I need to ensure that rke2 server can work properly.

**Comment by brandond** [comment]
Yes, you can get `/readyz` - however this requires authentication with an agent token. There is no anonymous healthcheck endpoint.

**Reply by huangyutongs** [reply]
> Yes, you can get `/readyz` - however this requires authentication with an agent token. There is no anonymous healthcheck endpoint.

Thanks for your reply,How is the token issued and how is it used for health check? Can you provide an example? Or is there any relevant documentation?

**Reply by brandond** [reply, marked as answer]
Sorry, it is `/readyz` under the `/v1-rke2` api prefix.

The agent token is the same token you use to join agents to the cluster? You can find it at `/var/lib/rancher/rke2/server/agent-token`. The portion you need is the credentials at the end:
```console
root@rke2-server-1:/# cat /var/lib/rancher/rke2/server/agent-token
K106283246fb75ff15c144ad0fc5d4a38d35cddbef2cd3e4acb1448824258c4f665::server:3bd367a56c4c2ee9e569d2766c88cfaa

root@rke2-server-1:/# curl -ksu node:3bd367a56c4c2ee9e569d2766c88cfaa https://rke2-server-1:9345/v1-rke2/readyz
ok
```

**Reply by huangyutongs** [reply]
> Sorry, it is `/readyz` under the `/v1-rke2` api prefix.
> 
> The agent token is the same token you use to join agents to the cluster? You can find it at `/var/lib/rancher/rke2/server/agent-token`. The portion you need is the credentials at the end:
> 
> ```
> root@rke2-server-1:/# cat /var/lib/rancher/rke2/server/agent-token
> K106283246fb75ff15c144ad0fc5d4a38d35cddbef2cd3e4acb1448824258c4f665::server:3bd367a56c4c2ee9e569d2766c88cfaa
> 
> root@rke2-server-1:/# curl -ksu node:3bd367a56c4c2ee9e569d2766c88cfaa https://rke2-server-1:9345/v1-rke2/readyz
> ok
> ```

Thank you very much, I can successfully access

**Comment by huangyutongs** [comment]
In addition, my k8s is v1.24.2+rke2r1
